lab_2_tokenize_by_bpe/main.py

```diff
@@ -81,6 +188,22 @@ def decode(
     :param end_of_word_token: an end-of-word token
     :return: decoded sequence
     """
+    if not (
+        isinstance(encoded_text, list) and
+        isinstance(vocabulary, dict) and
+        (isinstance(end_of_word_token, str) or end_of_word_token is None)
+    ):
+        return None
+
+    decoded = ''
+    voc_keys = list(vocabulary.keys())
+    for token in encoded_text:
+        symbol = voc_keys[token]
+        if symbol == end_of_word_token:
```

Comment:
Consider simply replacing all EoW tokens later when the resulting string is fully formed with a `replace` method

Reply:
done